examples/intraday_examples/lead_time_analysis.py

```diff
@@ -0,0 +1,312 @@
+    """This function uses the get_instance_curve to retrieve the instance curve data from all benchmark curves, then maps it into a dataframe.
+
+    Args:
+        instance_curves_names: A dictionary containing all instance curve names (the benchmark curves).
+        session: Session object to interact with WAPI.
+        issue_date: The user defined date of issue for the instance curve.
+
+
+    Returns:
+        df: A dataframe containing all instance curve data from the benchmark curves.
+    """
+    df = pd.DataFrame()
+    for curve_name in instance_curves_names.values():
+        instance_curve = get_instance_curve(curve_name=curve_name, session=session, issue_date=issue_date)
+        instance_curve.name = map_name_to_label(instance_curve.name)
+        df = pd.concat([df, instance_curve], axis=1)
+    return df
+
+
+def get_absolute_data(absolute_curves_names: str, session: Session, data_time: pd.Timestamp, time_delta: pd.Timedelta):
+    """This function uses the get_absolute_curve to retrieve the absolute curve data from all intraday curves (intraday or intraday_lastec), then maps it into a dataframe.
+
+    Args:
+        absolute_curves_names: A dictionary containing all absolute curve names (the intraday curves).
+        session: Session object to interact with WAPI.
+        data_time: The timestamp of the forecasted value.
+        time_delta: _description_
+
+    Returns:
+        df: A dataframe containing all absolute curve data from the intraday curves.
+    """
+
+    df = pd.DataFrame()
+
+    for curve_name in absolute_curves_names.values():
+        absolute_curve = get_absolute_curve(data_time=data_time, session=session, curve_name=curve_name,
+                                            delta_hours=time_delta)
+        absolute_curve.name = map_name_to_label(absolute_curve.name)
+        df = pd.concat([df, absolute_curve], axis=1)
+    return df
+
+
+def timezone_mapping(area: str) -> str:
+    """This function defines a timezone depending on the area input.
+
+    Args:
+        area: User defined area.
+
+    Returns:
+        str: The timezone displayed as a string.
+    """
+    if area.upper() == "TR":
+        return "trt"
+    elif "IE" in area.upper():
+        return "wet"
+    else:
+        return "cet"
+
+
+def map_name_to_label(curve_name: str):
+    """This function matches one name from the defined dictionary below with an inputted curve name and returns a more readable name.
+
+    Args:
+        curve_name: A curve name, for instance: "pro fr wnd ec00 mwh/h cet min15 f 2023-11-26T00:00:00+01:00"
+
+    Returns:
+        str: A name from the dictionary generated in this function. Can be used as column names in a df."
+    """
+    mapping = dict(spv="Spv production", wnd="Wind production", con="Consumption", rdl="Residual load")
+    return [mapping.get(name, 'Undefined') for name in mapping if name in curve_name][0]
+
+
+def str_to_datetime(time: pd.Timestamp, area: str) -> pd.Timestamp:
```

Comment:
I don't think we need this function as we're only using it in one place and it's a one-liner.

Reply:
True